### ISSUE TITLE
Replace 'preload' context with context calculated from the |as| attribute

### ISSUE THREAD

**Issue by briansmith** [whatwg/fetch#76]
```<link rel=preload>``` was changed to make the ```as``` attribute mandatory. This was done so that the browser can determine what context-sensitive parameters to supply to the fetch. We should take advantage of this by calculating a fetch context from the ```as``` attribute instead of having a ```preload``` fetch context. For example, ````<link rel=preload as=stylesheet>``` should use ```style``` as the fetch context. And, in particular, there should be no ```preload``` context. This will help ensure that all of the fetch's security checks, prioritization logic, caching header setting, etc. are consistent with what would happen in the non-preload case, so that a response from a preload will more exactly match the response that would be given from a non-preload fetch.

Note that a boolean "is this a preload" flag is still needed so that things like mixed content blocking can treat preloads specially.

Note that this would require a change to MIX, at least, because MIX should continue to block mixed-content preloads. (It already requires them to be blocked since 'preload' is not on the whitelist.) /cc @mikewest.

**Comment by igrigorik** [comment]
See: https://github.com/whatwg/fetch/issues/64#issuecomment-115700995. 

tl;dr: the plan is to replace "as" with type. The context will (and needs to) remain as "preload".

**Comment by briansmith** [comment]
I'm not convinced it is a good idea to continue using the "preload" context but I do agree that #64 subsumes this issue. I will comment in #64 later.
